### PR TITLE
Tl/mimedb

### LINKED ISSUE #2: Inline `mime-db` to avoid Artifact and JSON dependency
https://github.com/fonsp/MIMEs.jl/blob/43bf22fe44a60b0483ed6f30aa03d5781e30c85b/README.md?plain=1#L47

Instead of using Artifacts and JSON during package precompilation, we could this ourselves, and publish the parsed MIME database directly in our source code! The assumption is that this will give a performance benefit overall.

### PR DESCRIPTION
**Pull request by tlienart**
Separating the obtention of the mimedb to a script that can be executed manually that generates

* the raw db object
* the mime2ext and ext2mime objects

then serialises those, the module then deserialises those and calls them as before.

**Note**: I tried with having a single object serialised as a NamedTuple with all three sub-objects but it was much slower to load for some reason. 

In the future the `update.jl` script could be triggered by CI and automatically open PRs against the files here.

### CODE REVIEW

**Comment by fonsp** [comment]
Nice! But using Serialization will fix the package to a single Julia version. Maybe we can just use `string(data)` and write it a `.jl` file?

**Comment by tlienart** [comment]
done! (I use `include` to retrieve the content, maybe there's a better way but this works and is pretty fast so I didn't look much further)

**Comment by fonsp** [comment]
Cool! Do you have the new timings?

**Comment by fonsp** [comment]
Can we call it `.jlon`? 😅  This is essentially the Julia version of JSON

**Comment by tlienart** [comment]
lol, sure

File: mimedb/update.jl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    continue
+                end
+            end
+            _ext2mime[ex] = mime
+        end
+    end
+
+    @info "✏  writing to files ($mdb, $e2m, $m2e)..."
+    open(mdb, "w") do f
+        write(f, string(_mimedb))
```
Comment (tkf) [inline mimedb/update.jl]: Technically, I think you'd want to use `repr(_mimedb)` here (although it looks like their outputs are identical).

Also, `string(Dict(...))` and  `repr(Dict(...))` print vararg code:

```julia
julia> repr(Dict(:a => 1, :b => 2))
"Dict(:a => 1, :b => 2)"
```

I think this tries to compile a function with a very long argument (or maybe fallback to the interpreter?). Maybe it's a slight win for precompile time if we call `Dict(::Vector)` form

```julia
julia> d = Dict(:a => 1, :b => 2);

julia> string("Dict(", repr(collect(d)), ")")
"Dict([:a => 1, :b => 2])"
```

Comment (fonsp) [reply mimedb/update.jl]: We could also make our own struct with `exts, encoding, compressible, source`, and then write a bunch of constructors? Then we don't have to create all the small dictionaries, would that help?

**Comment by tkf** [comment]
Also, FYI, using `Symbol` could be a slight win for size and speed: https://github.com/tkf/MIMEFileExtensions.jl/pull/8

For example, if you store three `Dict`s with strings, I think you'd have three instances of `"text/plain"` in the precompile cache. With symbol, I think there is exactly one symbol and pointers to refer to it.

**Comment by tlienart** [comment]
So I did the `.jlon` thing, I tried playing with the Symbol and using a `Dict([...])` but got no significant changes in compilation time, the current is the fastest I was able to get. 

Maybe we can merge this and leave the precompilation question for another issue after more testing? (the results I got hover around 0.08s which seems fair for now)

**Comment by fonsp** [comment]
Awesome!! @tlienart I sent you a message on zulip about LiveServer x Pluto, take a look! @tkf thanks for joining forces and helping with the PR!

**Pull request merged**